src/openassetio-core/include/openassetio/hostApi/Manager.hpp

```diff
@@ -761,6 +761,164 @@ class OPENASSETIO_CORE_EXPORT Manager {
+   * be described as 'immediate children' of the location in question.
+   * This may not always be the case (say, for example there is some
+   * kind of 'task' structure in place too). Instead we use a request
+   * that asks for any 'shots' that relate to the chosen location. It is
+   * then up to the implementation of the manager to determine how that
+   * maps to its own data model. Hopefully this allows a host to work
+   * with a broader range of asset management systems, without providing
+   * any requirements of their structure or data model within the system
+   * itself.
+   *
+   * @{
+   */
+
+  /**
+   * Callback signature used for a successful entity relationship query.
+   */
+  using RelationshipSuccessCallback = std::function<void(std::size_t, EntityReferences)>;
+
+  /**
+   * Query entity references that are related to the input
+   * references by the relationship defined by a set of traits and
+   * their properties.
+   *
+   * This is an essential function in this API - as it is widely used
+   * to query other entities or organisational structure.
+   *
+   * @note Consult the documentation for the relevant relationship
+   * traits to determine if the order of entities in the inner lists
+   * of matching references is considered meaningful.
+   *
+   * If any relationship definition is unknown, then an empty list
+   * will be returned for that entity, and no errors will be
+   * raised.
+   *
+   * @param relationshipTraitsData The traits of the relationship to
+   * query.
+   *
+   * @param entityReferences A list of @ref entity_reference to query
+   * the specified relationship for.
+   *
+   * @param context The calling context.
+   *
+   * @param successCallback Callback that will be called for each
+   * successful relationship query. It will be given the
+   * corresponding index of the entity reference in
+   * @p entityReferences along with a list of entity references for
+   * entities that have the relationship specified by
+   * @p relationshipTraitsData. If there are no relations, the
+   * callback will receive an empty list. The callback will be called
+   * on the same thread that initiated the call to
+   * `getWithRelationship`.
+   *
+   * @param errorCallback Callback that will be called for each failed
+   * relationship query. It will be given the corresponding index of the
+   * entity reference in @p entityReferences along with a populated
+   * BatchElementError (see @ref BatchElementError.ErrorCode
+   * "ErrorCodes"). The callback will be called on the same thread that
+   * initiated the call to `getWithRelationship`.
+   *
+   * @param resultTraitSet A hint as to what traits the returned
+   * entities should have.
+   *
+   * @note The @ref trait_set of the queried relationship can be passed
+   * to @ref managementPolicy in order to determine if the manager
+   * handles relationships of that type.
+   */
+  void getWithRelationship(const TraitsDataPtr& relationshipTraitsData,
```

Comment:
[fix] Documenting offline chat. For consistency, `entityReferences` should come first.

Reply:
Rearranged in 4165c819